src/Optimization/hiopAlgFilterIPM.cpp

```diff
@@ -1055,7 +1055,7 @@ hiopSolveStatus hiopAlgFilterIPMQuasiNewton::run()
     /************************************************
      * update mu and other parameters
      ************************************************/
-    while(_err_log<=kappa_eps * _mu) {
+    if(_err_log<=kappa_eps * _mu) {
       //update mu and tau (fraction-to-boundary)
       bret = updateLogBarrierParameters(*it_curr, _mu, _tau, _mu, _tau);
       if(!bret) break; //no update is necessary
```

Comment:
why do we change this while loop to if? I think while is designed to reduce mu faster.
In addition, if we change to if, the break command on line 1061 may break the outer loop and give us unpredictable result.

Reply:
I switched back the `while` loop. It behaves like before with `elastic_mode` off, but when it's off, it only does one reduction of log barrier params. see commit #[55d8333](https://github.com/LLNL/hiop/pull/407/commits/55d83330df4024dfd40dcd4dfe3a1c25433ccf4d)